homeassistant/components/recorder/statistics.py

```diff
@@ -2065,7 +2096,11 @@ def _statistics_at_time(
 ) -> Sequence[Row] | None:
     """Return last known statistics, earlier than start_time, for the metadata_ids."""
     start_time_ts = start_time.timestamp()
-    stmt = _generate_statistics_at_time_stmt(table, metadata_ids, start_time_ts, types)
+    dialect_name = get_instance(hass).dialect_name
+    lateral_join_for_start_time = dialect_name == SupportedDialect.POSTGRESQL
+    stmt = _generate_statistics_at_time_stmt(
+        table, metadata_ids, start_time_ts, types, lateral_join_for_start_time
+    )
```

Comment:
Is the expectation that the `lateral_join_for_start_time` is set to True when we run with `POSTGRESQL` asserted in tests?